src/app/core/learning-object-module/file/file.service.ts

```diff
@@ -53,6 +59,198 @@ export class FileService {
+   * @param {string} route [Route to make request to]
+   * @returns
+   * @memberof LearningObjectService
+   */
+  private handleFileMetaRequests(
+    files: FileUploadMeta[],
+    route: string
+  ): Promise<string[]> {
+    const MAX_PER_REQUEST = 100;
+    const responses$: Promise<string[]>[] = [];
+    const completed$: Subject<boolean> = new Subject<boolean>();
+    const sendNextBatch$: Subject<void> = new Subject<void>();
+
+    const response = new Promise<string[]>((resolve, reject) => {
+      sendNextBatch$.pipe(takeUntil(completed$)).subscribe(() => {
+        const batch = files.splice(0, MAX_PER_REQUEST);
+        if (batch.length) {
+          this.handleFileMetaBatch(route, batch, responses$, sendNextBatch$);
+        } else {
+          this.handleFileMetaRequestQueueCompletion(completed$, responses$)
+            .then(resolve)
+            .catch(reject);
+        }
+      });
+    });
+    sendNextBatch$.next();
+    return response;
+  }
+
+  /**
+   * Handles making request to upload batch of file meta
+   *
+   * @private
+   * @param {string} route [Route to make request to]
+   * @param {FileUploadMeta[]} batch [Batch of file meta to be added]
+   * @param {Promise<string[]>[]} responses$ [List of response promises to append to]
+   * @param {Subject<void>} sendNextBatch$ [Observable used to signal that the next batch should be sent]
+   * @memberof LearningObjectService
+   */
+  private handleFileMetaBatch(
+    route: string,
+    batch: FileUploadMeta[],
+    responses$: Promise<string[]>[],
+    sendNextBatch$: Subject<void>
+  ) {
+    const response$ = this.http
+      .post(route,
+        { fileMeta: batch },
+        {
+          headers: this.headers,
+          withCredentials: true
+        })
+      .pipe(
+        catchError(this.handleError)
+      )
+      .toPromise()
+      .then((res: { fileMetaId: string[] }) => res.fileMetaId);
+    responses$.push(response$);
+    sendNextBatch$.next();
+  }
+
+  /**
+   * Handles completion of requests for all file metadata that was enqueued
+   *
+   * @private
+   * @param {Subject<boolean>} completed$ [Observable used to signal that all batches have been completed]
+   * @param {Promise<string[]>[]} responses$ [List of response promises to resolve]
+   * @returns
+   * @memberof LearningObjectService
+   */
+  private async handleFileMetaRequestQueueCompletion(
+    completed$: Subject<boolean>,
+    responses$: Promise<string[]>[]
+  ): Promise<string[]> {
+      completed$.next(true);
+      completed$.unsubscribe();
+      const fileIdsArrays = await Promise.all(responses$);
+      const fileIds = flattenDeep(fileIdsArrays);
+      return fileIds;
+    }
+
+  /**
+   * Handles downloading a file by opening the stream url in a new window
+   *
+   * @param {LearningObject.Material.File} file[The file to be downloaded]
+   * @memberof UploadComponent
+   */
+    async handleFileDownload(file: LearningObject.Material.File) {
```

Comment:
What component uses this? It looks like there aren't any references.

Reply:
Good catch, I ended up moving the function back to its original component file because it was directly used in the corresponding HTML file.

Reply:
If the function is using a file.routes.ts it should be moved to the file.service.ts but we just need to import the FileService into the component's constructor.

Reply:
Okay it's done ✅